package.json

```diff
@@ -44,11 +50,30 @@
 					"default": 0.3,
 					"scope": "application",
 					"description": "0 -> deterministic model, 1 -> Random model"
+				},
+				"general.frequency_penalty": {
+					"type": "number",
+					"default": 0.0,
+					"scope": "application",
+					"description": "0-1, Higher value -> Less chance of repeating same line"
+				},
+				"general.presence_penalty": {
+					"type": "number",
+					"default": 0.0,
+					"scope": "application",
+					"description": "0-1, Higher value -> Less chance of repeating same topic"
+				},
+				"general.stop": {
+					"type": "array",
+					"default": ["@"],
```

Comment:
did you forgot "def" and "class" ? My idea was to avoid having residual code after the completion. We can do so by stopping the generation before the next function.

Reply:
Sorry! Yes this was a mistake on my part, was about to comment.

Reply:
I have a new update I made just now that asks for user input on every ctrl+enter to enter an additional stop sequence (optionally), feels quite productive.

Reply:
I need to go to bed but write all issues you find and I can make a new pull request tomorrow :)

Reply:
```suggestion
					"default": ["@", "def", "class"],
```